src/AFQMC/Numerics/detail/HIP/Kernels/dot_wabn.hip.cpp

```diff
@@ -9,12 +9,10 @@
 // File created by: Fionn Malone, malone14@llnl.gov, LLNL
 ////////////////////////////////////////////////////////////////////////////////
 
-
-#include<cassert>
 #include <complex>
-#include<hip/hip_runtime.h>
 #include <thrust/complex.h>
-#include<hip/hip_runtime.h>
+#include <hip/hip_runtime.h>
+#include "Platforms/CUDA_legacy/uninitialized_array.hpp"
```

Comment:
Do not use this.
`#include <thrust/system/cuda/detail/core/util.h>`

Reply:
Ok. Why is it defined in CUDA_legacy?

Reply:
`uninitialized_array` was thought to be removed in CUDA by the CUDA_legacy code. So a copy of the file was put directly. But it is actually hidden instead of being removed. So in the new CUDA code, this header file copy is not used and the one from CUDA toolkit is used. ROCM has the same file.

Reply:
Thanks!

Reply:
CUDA legacy will be deleted.

Reply:
This header causes errors with the hip code. fatal error: 'cuda_occupancy.h' file not found. I can make the swap for the CUDA kernels only.